src/site/content/en/blog/prefers-color-scheme/index.md

```diff
@@ -0,0 +1,725 @@
+
+```html
+<!-- index.html -->
+<script>
+  // If `prefers-color-scheme` is not supported, fall back to light mode.
+  // In this case, the file will be downloaded with `highest` priority.
+  if (!window.matchMedia('(prefers-color-scheme)').matches) {
+    document.write('<link rel="stylesheet" href="/light.css"">');
+  }
+</script>
+<!-- The main stylesheet -->
+<link rel="stylesheet" href="/style.css">
+<!--
+  Conditionally either load the light or the dark stylesheet. The matching file
+  will be downloaded with `highest`, the non-matching file with `lowest`
+  priority. If the browser doesn't support `prefers-color-scheme`, the media
+  query is unknown and the files are downloaded with `lowest` priority (but
+  above I already force `highest` for my default light experience).
+-->
+<link rel="stylesheet" href="/dark.css" media="(prefers-color-scheme: dark)">
+<link rel="stylesheet" href="/light.css" media="(prefers-color-scheme: no-preference), (prefers-color-scheme: light)">
+```
+
+### Stylesheet architecture
+
+I make maximum use of [CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/var),
+this allows my generic `style.css` to be, well, generic,
+and all the customization happens in the two other files `dark.css` and `light.css`.
+Below you can see an excerpt of the actual styles, but it should suffice to convey the overall idea.
+I declare two variables, `--background-color` and `--color`
+that essentially create a *dark-on-light* and a *light-on-dark* baseline theme.
+
+```css
+/* light.css */
+:root {
+  --background-color: rgb(250, 250, 250);
+  --color: rgb(5, 5, 5);
+}
+```
+
+```css
+/* dark.css */
+:root {
+  --background-color: rgb(5, 5, 5);
+  --color: rgb(250, 250, 250);
+}
+```
+
+In my `style.css`, I then use these variables in the `body { … }` rule.
+As they are defined on the
+[`:root` CSS pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/:root)—a
+selector that in HTML represents the `<html>` element
+and is identical to the selector `html`, except that its specificity is
+higher—they cascade down, which serves me for declaring global CSS variables.
+
+```css
+/* style.css */
+:root {
+  color-scheme: dark light;
+}
+
+body {
+  background-color: var(--background-color);
+  color: var(--color);
+}
+```
+
+In the code sample above, you will probably have noticed a property
+[`color-scheme`](https://drafts.csswg.org/css-color-adjust-1/#propdef-color-scheme)
+with the space-separated value `light dark`.
+This tells the browser which color themes my app supports
+and allows it to activate special variants of the user agent stylesheet.
+This is useful to, for example, let the browser render form fields
+with a dark background and light text, adjust the scrollbars,
+or to enable a theme-aware highlight color.
+The exact details of `color-scheme` are specified in
+[CSS Color Adjustment Module Level&nbsp;1](https://drafts.csswg.org/css-color-adjust-1/).
+
+{% Aside 'note' %}
```

Comment:
Not sure about the look of the two `aside`s following each other. Perhaps the content of the first one could be moved inline to the preceding text.

Reply:
Ungrouped the two by reflowing a bit in https://github.com/GoogleChrome/web.dev/pull/1021/commits/73edd13d82997519504aa88fd39540070b2c0821.